go/cmd/vtctldclient/command/throttler.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"github.com/spf13/cobra"
+
+	"vitess.io/vitess/go/cmd/vtctldclient/cli"
+
+	vtctldatapb "vitess.io/vitess/go/vt/proto/vtctldata"
+)
+
+const (
+	// noValueIndicator helps to differentiate between a "no value given in --custom-query hence don't change it"
+	// vs. a "change --custom-query to be empty" scenarios.
+	noValueIndicator = "~"
+)
```

Comment:
ah, now i get what you're going for here. because `cobra` uses `pflag` under the hood, you can actually check whether a flag was unspecified vs explicitly set to the empty/zero value: https://pkg.go.dev/github.com/spf13/pflag#FlagSet.Changed

so in your commandUpdateThrottlerConfig, you can do something like:

```go
switch {
case: !cmd.Flags().Changed("custom-query"):
    // flag was not specified
case updateThrottlerConfigOptions.CustomQuery == "":
    // flag was explicitly set to empty
case ...:
default:
}
```

Reply:
so I can tell the difference at the time of parsing the flags, but I'm still unable to pass on this information in `UpdateThrottlerConfigRequest`.

Alternatively, I can add a `--clear-custom-query` (bool) flag, mutually exclusive to `--custom-query` and have that parameter all the way down in proto.